src/WhiteLibrary/keywords/application.py

```diff
@@ -26,6 +27,25 @@ def launch_application(self, sut_path, args=None):
         else:
             self.state.app = Application.Launch(sut_path)
 
+    def _attach_application(self, sut_identifier):
+        if self.state._attach_application_timeout == 0:
+            return Application.Attach(sut_identifier)
+
+        hack = {"sut": None}
```

Comment:
nonlocal doesnt work in py2, walrus operator for wait condition is only for py3.8, inner functions cannot modify outerscope primitive variables but can modify object / dict properties so having it like this is only portable way to get around getting the Application reference in the `search_application()` with a single call :(